pkg/secretless/plugin/README.md

```diff
@@ -0,0 +1,94 @@
+
+## Plugin Metadata
+
+Regardless of plugin type, each plugin must supply Secretless with some essential metadata.
+To do this, each plugin must implement the `PluginInfo` function. This top level function is
+always required and it returns basic information about your plugin. Its signature is:
+
+```go
+func PluginInfo() map[string]string
+```
+
+The returned map must have the following keys:
+
+- `version`: The version of the plugin itself.  This allows plugin authors to
+  version the plugins they write.
+- `pluginAPIVersion`: The version of the Secretless plugin API your plugin is
+  written for.  This allows the Secretless plugin API to change over time
+  without breaking plugins.
+  <!-- TODO: how can a plugin dev find the appropriate version to use? -->
+- `type`: This must be a supported plugin type. Currently, it must be
+  either the string `"connector.tcp"` or the string `"connector.http"`.
+- `id`: A short, clear, unique name, for use in logs and the `secretless.yml`
+  config file.  Allowed characters are: lowercase letters, `_`, `:`, `-`, and
+  `~`.
+- `description`: A short summary of the plugin, not to exceed 100 characters.
+  This may be used in the future by the Secretless command line tool to list
+  available plugins.
+
+## External Plugin Basics
+
+When running Secretless with external plugins, you may leverage some special
+command-line flags when starting Secretless:
+
+- `-p` flag: Specifies the directory that the external plugins shared library (".so") files
+  live in. Defaults to `/usr/local/lib/secretless`. Sub-directory traversal
+  is not supported at this time.
+- `-s` flag: Refers to a file which contains sha256sum plugin checksums for verifying the plugins.
+
+When Secretless starts, it:
+- Checks for available external plugins (eg ".so" files) in the plugin directory
+- Verifies external plugin checksums (if a checksum file was provided on start)
+- Loads the external plugin. That is, for each plugin file, Secretless:
+  - Opens the [Go plugin](https://golang.org/pkg/plugin/) file
+  - Parses `PluginInfo` for [plugin metadata](#plugin-metadata)
+  - Verifies that the plugin type supplied in `PluginInfo` is supported
+  - Loads the plugin into the list of plugins to run
+
+From there, the startup process continues and external plugins are treated the same
+as internal plugins.
+
+### Building the Shared Library File
+To build your plugin's shared library (`.so`) file, you follow the [instructions](https://golang.org/pkg/plugin/) for building Go plugins. That is, you run a command like:
+```
+go build -buildmode=plugin -o=/path/to/my-plugin.so my_plugin.go
+```
+
+This compiles your plugin code into a `.so` file. Once you've done this,
+you'll place the `.so` file in the `/usr/local/lib/secretless` directory in the
+container where Secretless will run (or in another directory you specify using the `-p` flag).
+That's all you have to do. The `.so` files are self-contained, with any dependencies
```

Comment:
I'm fine with making this change here, but I agree it would be good to discuss where it's appropriate to use a conversational tone - maybe it's appropriate in tutorials and blog posts, but not in reference material or guides?

Reply:
That is a good way to approach it!